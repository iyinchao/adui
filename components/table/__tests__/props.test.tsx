/* eslint-disable */
import * as React from "react"
import { mount } from "enzyme"
import TestRenderer from "react-test-renderer"
import sinon from "sinon"
import Table from ".."

window.ResizeObserver =
  window.ResizeObserver ||
  jest.fn().mockImplementation(() => ({
    disconnect: jest.fn(),
    observe: jest.fn(),
    unobserve: jest.fn(),
  }))

const dataSource = [
  {
    one: "阿萨德",
    two: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    four: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    three: "阿萨德阿萨德阿萨德阿萨德阿萨德",
  },
  {
    one: "阿萨德",
    two: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    four: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    three: "阿萨德阿萨德阿萨德阿萨德阿萨德",
  },
  {
    one: "阿萨德",
    two: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    four: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    three: "阿萨德阿萨德阿萨德阿萨德阿萨德",
  },
  {
    one: "阿萨德",
    two: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    four: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    three: "阿萨德阿萨德阿萨德阿萨德阿萨德",
  },
  {
    one: "阿萨德",
    two: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    four: "阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德",
    three: "阿萨德阿萨德阿萨德阿萨德阿萨德",
  },
]

export interface IBaseObject {
  [key: string]: any
}

describe("Table", () => {
  beforeAll(() => {
    /**
     * https://stackoverflow.com/a/40006663
     * 通过 mock `getBoundingClientRect`，之后运行的所有 `getBoundingClientRect` 都会返回这同一个值。
     */
    Element.prototype.getBoundingClientRect = jest.fn(() => {
      return {
        width: 500,
        height: 600,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
        x: 0,
        y: 0,
        toJSON: () => {},
      }
    })
  })

  it("headerAffixed 检测", () => {
    const wrapper = mount(
      <Table dataSource={dataSource} headerAffixed>
        <Table.Column
          dataIndex="one"
          title="第一项"
          popover="表头提示内容"
          width="100px"
          fixed="left"
        />
        <Table.Column
          fixed="right"
          dataIndex="two"
          title="第er项"
          width={300}
        />
        <Table.Column
          dataIndex="s"
          title="第si项"
          width="10%"
          align="center"
          render={() => "这些撒大声地"}
        />
        <Table.Column dataIndex="three" title="第san项" />
        <Table.Column
          dataIndex="four"
          title="第si项"
          width={300}
          align="right"
        />
      </Table>
    )
    // @ts-ignore
    global.innerWidth = 500
    // @ts-ignore
    global.dispatchEvent(new Event("resize"))
    // @ts-ignore
    global.dispatchEvent(new Event("scroll"))
    wrapper.instance().handleMainTableScroll()
    wrapper.instance().componentWillUnmount()
  })

  // it("height prop 验证", () => {
  //   const stub = sinon.stub(console, 'error')
  //   const wrapper = mount(
  //     <Table
  //       dataSource={dataSource}
  //       height="500px"
  //     >
  //       <Table.Column
  //         dataIndex="one"
  //         title="第一项"
  //         popover="表头提示内容"
  //         width={100}
  //         fixed="left"
  //       />
  //     </Table>
  //   )
  //   expect(stub.calledOnce).toBe(true)
  //   stub.restore()
  // })

  it("height 与 headerAffixed 不能同时使用 验证", () => {
    const stub = sinon.stub(console, "error")
    mount(
      <Table dataSource={dataSource} height={500} headerAffixed>
        <Table.Column
          dataIndex="one"
          title="半角括号外(半角括号内测试)"
          popover="表头提示内容"
          width={100}
          fixed="left"
        />
      </Table>
    )
    expect(stub.calledOnce).toBe(true)
    stub.restore()
  })

  it("height 检测", () => {
    const wrapper = mount(
      <Table dataSource={dataSource} height={200}>
        <Table.Column
          dataIndex="one"
          title="第一项"
          popover="表头提示内容"
          width={100}
          fixed="left"
        />
        <Table.Column
          fixed="right"
          dataIndex="two"
          title="第er项"
          width={300}
        />
        <Table.Column
          dataIndex="s"
          title="第si项"
          width={300}
          align="center"
          render={() => "这些撒大声地"}
        />
        <Table.Column dataIndex="three" title="第san项" />
        <Table.Column
          dataIndex="four"
          title="第si项"
          width={300}
          align="right"
        />
      </Table>
    )
    // @ts-ignore
    global.innerWidth = 500
    // @ts-ignore
    global.innerHeight = 500
    // @ts-ignore
    global.dispatchEvent(new Event("resize"))
    // @ts-ignore
    global.dispatchEvent(new Event("scroll"))
    wrapper.instance().handleMainTableScroll()
  })

  // it("Column width prop 验证", () => {
  //   const stub = sinon.stub(console, 'error')
  //   const wrapper = mount(
  //     <Table
  //       dataSource={dataSource}
  //     >
  //       <Table.Column
  //         dataIndex="one"
  //         title="第一项"
  //         width={[]}
  //       />
  //     </Table>
  //   )
  //   expect(stub.calledOnce).toBe(true)
  //   stub.restore()
  // })

  it("Column width 在 fixed 时必须设置 验证", () => {
    const stub = sinon.stub(console, "error")
    mount(
      <Table dataSource={dataSource}>
        <Table.Column dataIndex="one" title="第一项" fixed="left" />
      </Table>
    )
    expect(stub.calledOnce).toBe(true)
    stub.restore()
  })

  it("loading 快照", () => {
    const wrapper = TestRenderer.create(
      <Table loading dataSource={dataSource}>
        <Table.Column dataIndex="one" title="第一项" fixed="left" />
      </Table>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it("合并单元格快照", () => {
    const wrapper = TestRenderer.create(
      <Table
        dataSource={dataSource}
        getCellProps={(_, __, rowIndex, colIndex) => {
          if (rowIndex === 0 && colIndex === 0) {
            return {
              rowSpan: 2,
              colSpan: 2,
            }
          }
          if (rowIndex === 3 && colIndex === 2) {
            return {
              rowSpan: 2,
              colSpan: 1,
            }
          }
          if (rowIndex === 3 && colIndex === 3) {
            return {
              rowSpan: 2,
              colSpan: 1,
            }
          }
          if (rowIndex === 2 && colIndex === 0) {
            return {
              rowSpan: 1,
              colSpan: 1,
            }
          }
          return {}
        }}
      >
        <Table.Column dataIndex="one" title="第一项" fixed />
        <Table.Column dataIndex="two" title="第er项" fixed />
        <Table.Column dataIndex="three" title="第3项" align="center" />
        <Table.Column
          dataIndex="four"
          title="第四项"
          align="center"
          fixed="right"
        />
      </Table>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
