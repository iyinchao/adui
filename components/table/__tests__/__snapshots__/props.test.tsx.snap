// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table loading 快照 1`] = `
<div
  className="adui-table-wrapper adui-table-small adui-table-loading"
>
  <div
    style={
      Object {
        "position": "relative",
        "top": "40px",
        "zIndex": "calc(var(--z-index-affix) + 2)",
      }
    }
  >
    <div
      className="adui-table-progress"
    >
      <i />
      <i />
    </div>
  </div>
  <div
    className="adui-table-tables"
    style={
      Object {
        "maxHeight": undefined,
        "paddingTop": "",
      }
    }
  >
    <div
      className="adui-table-mainTable"
      onScroll={[Function]}
      role="table"
    >
      <div
        className="adui-table-thead"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "height": "40px",
          }
        }
      >
        <div
          className="adui-table-th adui-table-th_left"
          data-column={0}
          style={
            Object {
              "flex": "100 0 auto",
              "left": 0,
              "maxWidth": "",
              "right": undefined,
              "textAlign": undefined,
              "width": 100,
            }
          }
        >
          <div
            className="adui-table-cell"
            style={
              Object {
                "justifyContent": undefined,
              }
            }
          >
            第一项
          </div>
        </div>
      </div>
      <div
        className="adui-table-tbody"
      >
        <div
          className="adui-table-tr"
          data-row={0}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left"
            data-column={0}
            data-row={0}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
              </div>
            </div>
          </div>
        </div>
        <div
          className="adui-table-tr"
          data-row={1}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left"
            data-column={0}
            data-row={1}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
              </div>
            </div>
          </div>
        </div>
        <div
          className="adui-table-tr"
          data-row={2}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left"
            data-column={0}
            data-row={2}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
              </div>
            </div>
          </div>
        </div>
        <div
          className="adui-table-tr"
          data-row={3}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left"
            data-column={0}
            data-row={3}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
              </div>
            </div>
          </div>
        </div>
        <div
          className="adui-table-tr"
          data-row={4}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left"
            data-column={0}
            data-row={4}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Table 合并单元格快照 1`] = `
<div
  className="adui-table-wrapper adui-table-small"
>
  <div
    style={
      Object {
        "position": "relative",
        "top": "40px",
        "zIndex": "calc(var(--z-index-affix) + 2)",
      }
    }
  />
  <div
    className="adui-table-tables adui-table-hasFixedRightColumns"
    style={
      Object {
        "maxHeight": undefined,
        "paddingTop": "",
      }
    }
  >
    <div
      className="adui-table-mainTable"
      onScroll={[Function]}
      role="table"
    >
      <div
        className="adui-table-thead"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "height": "40px",
          }
        }
      >
        <div
          className="adui-table-th adui-table-th_left"
          data-column={0}
          style={
            Object {
              "flex": "100 0 auto",
              "left": 0,
              "maxWidth": "",
              "right": undefined,
              "textAlign": undefined,
              "width": 100,
            }
          }
        >
          <div
            className="adui-table-cell"
            style={
              Object {
                "justifyContent": undefined,
              }
            }
          >
            第一项
          </div>
        </div>
        <div
          className="adui-table-th adui-table-th_left"
          data-column={1}
          style={
            Object {
              "flex": "100 0 auto",
              "left": 0,
              "maxWidth": "",
              "right": undefined,
              "textAlign": undefined,
              "width": 100,
            }
          }
        >
          <div
            className="adui-table-cell"
            style={
              Object {
                "justifyContent": undefined,
              }
            }
          >
            第er项
          </div>
        </div>
        <div
          className="adui-table-th"
          data-column={2}
          style={
            Object {
              "flex": "100 0 auto",
              "left": undefined,
              "maxWidth": "",
              "right": undefined,
              "textAlign": "center",
              "width": 100,
            }
          }
        >
          <div
            className="adui-table-cell"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            第3项
            <div
              className="adui-table-resizer"
              onMouseDown={[Function]}
              role="none"
            />
          </div>
        </div>
        <div
          className="adui-table-th adui-table-th_right"
          data-column={3}
          style={
            Object {
              "flex": "100 0 auto",
              "left": undefined,
              "maxWidth": "",
              "right": 0,
              "textAlign": "center",
              "width": 100,
            }
          }
        >
          <div
            className="adui-table-cell"
            style={
              Object {
                "justifyContent": "center",
              }
            }
          >
            第四项
          </div>
        </div>
      </div>
      <div
        className="adui-table-tbody"
      >
        <div
          className="adui-table-tr"
          data-row={0}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left adui-table-td_combined"
            data-column={0}
            data-row={0}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
                <div
                  className="adui-table-cell_combined"
                  style={
                    Object {
                      "textAlign": undefined,
                    }
                  }
                >
                  阿萨德
                </div>
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_left"
            data-column={1}
            data-row={0}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td"
            data-column={2}
            data-row={0}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_right"
            data-column={3}
            data-row={0}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
        </div>
        <div
          className="adui-table-tr"
          data-row={1}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left"
            data-column={0}
            data-row={1}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_left"
            data-column={1}
            data-row={1}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td"
            data-column={2}
            data-row={1}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_right"
            data-column={3}
            data-row={1}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
        </div>
        <div
          className="adui-table-tr"
          data-row={2}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left adui-table-td_combined"
            data-column={0}
            data-row={2}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
                <div
                  className="adui-table-cell_combined"
                  style={
                    Object {
                      "height": "calc(100% - 1px)",
                      "textAlign": undefined,
                      "width": "calc(100% - 1px)",
                    }
                  }
                >
                  阿萨德
                </div>
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_left"
            data-column={1}
            data-row={2}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td"
            data-column={2}
            data-row={2}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_right"
            data-column={3}
            data-row={2}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
        </div>
        <div
          className="adui-table-tr"
          data-row={3}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left"
            data-column={0}
            data-row={3}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_left"
            data-column={1}
            data-row={3}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_combined"
            data-column={2}
            data-row={3}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德
                <div
                  className="adui-table-cell_combined"
                  style={
                    Object {
                      "textAlign": "center",
                      "width": "calc(100% - 1px)",
                    }
                  }
                >
                  阿萨德阿萨德阿萨德阿萨德阿萨德
                </div>
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_right adui-table-td_combined"
            data-column={3}
            data-row={3}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
                <div
                  className="adui-table-cell_combined"
                  style={
                    Object {
                      "textAlign": "center",
                      "width": "calc(100% - 1px)",
                    }
                  }
                >
                  阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="adui-table-tr"
          data-row={4}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="row"
          style={Object {}}
        >
          <div
            className="adui-table-td adui-table-td_left"
            data-column={0}
            data-row={4}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_left"
            data-column={1}
            data-row={4}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": 0,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": undefined,
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td"
            data-column={2}
            data-row={4}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
          <div
            className="adui-table-td adui-table-td_right"
            data-column={3}
            data-row={4}
            role="cell"
            style={
              Object {
                "flex": "100 0 auto",
                "left": undefined,
                "maxWidth": "",
                "right": undefined,
                "width": 100,
              }
            }
          >
            <div
              className="adui-table-cell"
              style={
                Object {
                  "alignItems": undefined,
                  "justifyContent": "center",
                }
              }
            >
              <div
                className="adui-table-cellInner"
              >
                阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德阿萨德
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
